### ISSUE TITLE
Add new category form: more a11y improvements

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#2634]
Splitting this out from #2588, see #2572.

A couple improvements worth considering for the "Add new category" mini-form in the sidebar:

1
When adding a new category, the submit button gets disabled to avoid multiple submissions. Elements that have focus should never gets disabled. When using a keyboard and activating the button, it gets disabled (see `disabled={ adding }`) thus causing a focus loss. Ideally, the button should just noop. This applies to all other similar cases.

2
Add a Speak message when a term is successfully added (maybe also on failure?)

**Comment by karmatosed** [comment]
If I am correct, none of these have design implications? I would +1 anything being just implemented if it doesn't need mocking up/change experience for all and would improve the experience for all.

**Comment by afercia** [comment]
Correct! The only (barely noticeable) difference would be the button not getting the "disabled" style.
